NetSDK/Services/SegmentFetcher/Classes/SelfRefreshingSegment.cs

```diff
@@ -58,26 +66,24 @@ private void RefreshSegment()
             while (true)
             {
                 try
-                {            
-                    var response = segmentChangeFetcher.Fetch(name, change_number);
+                {
+                    var changeNumber = segmentCache.GetChangeNumber(name);
+                    var response = segmentChangeFetcher.Fetch(name, changeNumber);
                     if (response == null)
                     {
                         return;
                     }
-                    if (change_number >= response.till)
+                    if (changeNumber >= response.till)
                     {
                         gates.SegmentIsReady(name);
                         return;
```

Comment:
Doesn't make sense the order here. Why we do:
1- var response = segmentChangeFetcher.Fetch(name, changeNumber);
2- if (changeNumber >= response.till) return

Am I losing something?